packages/mdc-textfield/index.js

```diff
@@ -188,6 +214,19 @@ class MDCTextField extends MDCComponent {
             this.bottomLine_.unlisten(evtType, handler);
           }
         },
+        getWidth: () => this.root_.offsetWidth,
+        getHeight: () => this.root_.offsetHeight,
+        getLabelWidth: () => {
+          const labelElement = this.root_.querySelector(strings.LABEL_SELECTOR);
```

Comment:
Is this safe? We treat label as optional overall...but I suppose it's required for outlined text fields?

(I can't actually remember in what cases label isn't provided, off the top of my head... Full-width?)

I'm also wondering whether this belongs in the label subcomponent's adapter, though that might not be possible without making it more onerous to use from this module.

Reply:
Hmm yeah it makes more sense to get the label width from the label's adapter, even if it means it'll have to be passed to the label foundation, then the main foundation, then the outline foundation.